signer/tests/integration/withdrawal_reject.rs

```diff
@@ -0,0 +1,677 @@
+
+    let validation_result = reject_withdrawal_tx.validate(&ctx, &req_ctx).await;
+    match validation_result.unwrap_err() {
+        Error::WithdrawalRejectValidation(ref err) => {
+            assert_eq!(err.error, WithdrawalRejectErrorMsg::RequestMissing)
+        }
+        err => panic!("unexpected error during validation {err}"),
+    }
+
+    testing::storage::drop_db(db).await;
+}
+
+/// For this test we check that the `RejectWithdrawalV1::validate` function
+/// returns a withdrawal validation error with a RequestNotFinal message
+/// when bitmap in the transaction does not match what our records would
+/// create for the bitmap.
+#[tokio::test]
+async fn reject_withdrawal_validation_not_final() {
+    // Normal: this generates the blockchain as well as a transaction
+    // sweeping out the funds for a withdrawal request. This is just setup
+    // and should be essentially the same between tests.
+    let db = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let (rpc, faucet) = regtest::initialize_blockchain();
+
+    let amount = 1_000_000;
+    let test_signer_set = TestSignerSet::new(&mut rng);
+    let deposit_amounts = DepositAmounts { amount, max_fee: amount / 2 };
+    let mut setup =
+        TestSweepSetup2::new_setup(test_signer_set.clone(), &faucet, &[deposit_amounts]);
+
+    setup.submit_sweep_tx(rpc, faucet, true);
+
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_stacks_client()
+        .with_mocked_emily_client()
+        .build();
+
+    let public_keys = test_signer_set
+        .keys
+        .iter()
+        .cloned()
+        .collect::<BTreeSet<PublicKey>>();
+    ctx.state().update_current_signer_set(public_keys);
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    backfill_bitcoin_blocks(&db, rpc, &setup.sweep_block_hash().unwrap()).await;
+
+    // Normal: we take the sweep transaction as is from the test setup and
+    // store it in the database.
+    setup.store_sweep_tx(&db).await;
+
+    // Normal: we need to store a row in the dkg_shares table so that we
+    // have a record of the scriptPubKey that the signers control.
+    setup.store_dkg_shares(&db).await;
+
+    // Normal: the request and how the signers voted needs to be added to
+    // the database. Here the bitmap in the withdrawal request object
+    // corresponds to how the signers voted.
+    setup.reject_withdrawal_request();
+    setup.store_withdrawal_request(&db).await;
+    setup.store_withdrawal_decisions(&db).await;
+
+    // Generate an insufficient number of blocks then backfill the DB
+    let mut hashes = faucet.generate_blocks(5);
```

Comment:
For this test, after we update the validation logic to reflect the first proposed change in section 3.1 of https://github.com/stacks-network/sbtc/issues/1363, we just need to change the `5` here to `WITHDRAWAL_BLOCKS_EXPIRY - 1`.

Reply:
Done.